Mathlib/Algebra/Category/ModuleCat/Topology/Basic.lean

```diff
@@ -0,0 +1,457 @@
+
+open Limits
+
+/-- The cone of topological modules associated to a cone over the underlying modules, where
+the cone point is given the induced topology. This is limiting when the given cone is. -/
+def ofCone {J : Type*} [Category J] {F : J ⥤ TopModuleCat R}
+    (c : Cone (F ⋙ forget₂ _ (ModuleCat R))) : Cone F where
+  pt := induced c.π.app
+  π :=
+  { app := fromInduced c.π.app,
+    naturality {X Y} f := by ext x; exact congr($(c.π.naturality f).hom x) }
+
+/-- Given a limit cone over the underlying modules, equipping the cone point with
+the induced topology gives a limit cone in `TopModuleCat R`. -/
+def isLimit {J : Type*} [Category J] {F : J ⥤ TopModuleCat R}
+    {c : Cone (F ⋙ forget₂ _ (ModuleCat R))} (hc : IsLimit c) :
+    IsLimit (ofCone c) where
+  lift s := ofHom (Y := (ofCone c).pt) ⟨(hc.lift ((forget₂ _ _).mapCone s)).hom, by
+    rw [continuous_iff_coinduced_le]
+    refine le_iInf fun i ↦ ?_
+    rw [coinduced_le_iff_le_induced, induced_compose, ← continuous_iff_le_induced]
+    show Continuous (X := s.pt) (Y := F.obj i)
+      (hc.lift ((forget₂ _ (ModuleCat R)).mapCone s) ≫ c.π.app i).hom
+    rw [hc.fac]
+    exact (s.π.app i).hom.2⟩
+  fac s i := by ext x; exact congr($(hc.fac ((forget₂ _ _).mapCone s) i).hom x)
+  uniq s m H := by
+    ext x
+    refine congr($(hc.uniq ((forget₂ _ _).mapCone s) ((forget₂ _ _).map m) fun j ↦ ?_).hom x)
+    ext y
+    exact congr($(H j).hom y)
+
+instance hasLimit_of_hasLimit_forget₂ {J : Type*} [Category J] {F : J ⥤ TopModuleCat.{v} R}
+    [HasLimit (F ⋙ forget₂ _ (ModuleCat.{v} R))] : HasLimit F :=
+  ⟨_, isLimit (limit.isLimit _)⟩
+
+instance {J : Type*} [Category J] [HasLimitsOfShape J (ModuleCat.{v} R)] :
+    HasLimitsOfShape J (TopModuleCat.{v} R) where
+  has_limit _ := hasLimit_of_hasLimit_forget₂
+
+instance : HasLimits (TopModuleCat.{v} R) where
+  has_limits_of_shape _ _ := ⟨fun _ ↦ hasLimit_of_hasLimit_forget₂⟩
+
+instance {J : Type*} [Category J] {F : J ⥤ TopModuleCat.{v} R}
+    [HasLimit (F ⋙ forget₂ _ (ModuleCat.{v} R))]
+    [PreservesLimit (F ⋙ forget₂ _ (ModuleCat.{v} R)) (forget _)] :
+    PreservesLimit F (forget₂ _ TopCat) :=
+  preservesLimit_of_preserves_limit_cone (isLimit (limit.isLimit _))
+    (TopCat.isLimitConeOfForget (F := F ⋙ forget₂ _ TopCat)
+      ((forget _).mapCone (getLimitCone (F ⋙ forget₂ _ (ModuleCat.{v} R))).1:)
+      (isLimitOfPreserves (forget (ModuleCat R)) (limit.isLimit _)))
+
+instance {J : Type*} [Category J]
+    [HasLimitsOfShape J (ModuleCat.{v} R)]
+    [PreservesLimitsOfShape J (forget (ModuleCat.{v} R))] :
+    PreservesLimitsOfShape J (forget₂ (TopModuleCat.{v} R) TopCat) where
+
+instance : PreservesLimits (forget₂ (TopModuleCat.{v} R) TopCat.{v}) where
+
+end Limit
+
+section Adjunction
+
+instance (priority := low) {R : Type*} [TopologicalSpace R] {A : Type*} [Add A] [SMul R A] :
+    letI := moduleTopology R A; IsModuleTopology R A :=
+  letI := moduleTopology R A; ⟨rfl⟩
```

Comment:
Is this useful? Why is it low prio? Maybe add some comments?

Reply:
Note to self: This should be moved.

This is low priority because AFAIK the topology is not part of the key on the discriminant tree. That is, every instance search for `IsModuleTopology` will trigger this and try to check if the topology is defeq to `moduleTopology R A`.